lib/datadog/tracing/contrib/rails/patcher.rb

```diff
@@ -90,10 +73,6 @@ def after_intialize(app)
               # Finish configuring the tracer after the application is initialized.
               # We need to wait for some things, like application name, middleware stack, etc.
               setup_tracer
-
-              # `after_initialize` will respect the configuration from `config/initializers/datadog.rb`
-              # and add tags to `::ActiveSupport::TaggedLogging`
-              add_tags_to_logger(app) if Datadog.configuration.tracing.log_injection
```

Comment:
We don't have to try to inject tag logs at the `after_initialize` stage anymore because we always inject a safe injection proc early, at `before_initialize`.
This simplifies our logic quite a bit.